services/ingestion/etl/euinvest/csv/test/unit/lib/__snapshots__/transform.spec.js.snap

```diff
@@ -178,3 +178,582 @@ The system will help resolve flooding problems and provide an outlet for unused
   "type": Array [],
 }
 `;
+
+exports[`EU Invest CSV transformer Snapshot project_locations: can handle multiple regions 1`] = `
+Object {
+  "action": "",
+  "budget": Object {
+    "eu_contrib": Object {
+      "currency": "",
+      "raw": "",
+      "value": 0,
+    },
+    "funding_area": Array [],
+    "mmf_heading": "",
+    "other_contrib": Object {
+      "currency": "",
+      "raw": "",
+      "value": 0,
+    },
+    "private_fund": Object {
+      "currency": "",
+      "raw": "",
+      "value": 0,
+    },
+    "public_fund": Object {
+      "currency": "",
+      "raw": "",
+      "value": 0,
+    },
+    "total_cost": Object {
+      "currency": "",
+      "raw": "",
+      "value": 0,
+    },
+  },
+  "call_year": "",
+  "complete": true,
+  "description": "To help counteract climate change, the EU is backing the German town of Emmendingen’s ‘Emmendingen Energy House’ campaign. The campaign supports residents wanting to cut energy use by renovating their homes, including through technical consultancy and a network of building modernisation professionals.
+
+The ‘Emmendingen Energy House’ campaign facilitates renovation of older residential buildings in order to tackle climate change
+Campaign tools include awareness raising among residents, a local consultancy programme and a network of building modernisation professionals
+The campaign is funded by the EU to the tune of €285,300 and by the state of Baden-Württemberg",
+  "ec_priorities": Array [],
+  "media": Array [
+    Object {
+      "meta": Object {
+        "description": "",
+        "mime_type": "",
+        "type": "visual",
+      },
+      "name": "https://europa.eu/investeu/sites/investeu/files/de_bawu_banner_985x270.png",
+      "url": "https://europa.eu/investeu/sites/investeu/files/de_bawu_banner_985x270.png",
+    },
+    Object {
+      "meta": Object {
+        "description": "Stadt Emmendingen",
+        "mime_type": "",
+        "type": "visual",
+      },
+      "name": "https://europa.eu/investeu/sites/investeu/files/de_bawu_homepage_768x432.png",
+      "url": "https://europa.eu/investeu/sites/investeu/files/de_bawu_homepage_768x432.png",
+    },
+  ],
+  "programme_name": "",
+  "project_id": "518",
+  "project_locations": Array [
+    Object {
+      "address": "",
+      "centroid": null,
+      "country_code": "DE",
+      "location": null,
+      "nuts": Array [],
+      "postal_code": "",
+      "region": "Emmendingen",
```

Comment:
`Emmendingen` is the town, `Baden-Württemberg` the region.

Same with `Albstadt-Tailfingen` (town).

I guess that if there are 3 parts in the location: part[0] = town, part[1] = region, part[2] = country code

Just an assumption from what I see

Reply:
Thanks for noticing! It turns out this part will need even more improvements.

Reply:
I moved existing logic to the lib module and corrected the known town/region cases.
I'll add a few more tests tomorrow in the lib to cover for a few more known edge cases.